discordBot.js

```diff
@@ -43,6 +43,47 @@ class DiscordBot {
   stop() {
     return this.client.destroy();
   }
+
+  // send files to discord channel given the file path and channel
+  sendFile(filePath, channelId){
+    this.client.on('ready', () => {
+      var attachment = new MessageAttachment(filePath);
```

Comment:
again, no need to have this.client.on() for the discordbot call
could just call the api
```
var attachment = new MessageAttachment(filePath);
return this.client.channels.cache.get(channelId).send(attachment);
```

Reply:
it is needed otherwise the send function channels.cache.....send(attachment) wont work because it would be empty

Reply:
I don't how you wrote the test on your side, but the `client.on('ready')` is apparently a callback method being fired when the the user is login.

Reply:
the channel empty might be sth related to how discordBot deal with cache updating, need more investigate on that.